HRIS.Services/Services/FieldCodeService.cs

```diff
@@ -11,43 +12,45 @@ public class FieldCodeService : IFieldCodeService
 {
     private readonly IUnitOfWork _db;
     private readonly IFieldCodeOptionsService _fieldCodeOptionsService;
+    private readonly AuthorizeIdentity _identity;
 
-    public FieldCodeService(IUnitOfWork db, IFieldCodeOptionsService fieldCodeOptionsService)
+    public FieldCodeService(IUnitOfWork db, IFieldCodeOptionsService fieldCodeOptionsService, AuthorizeIdentity identity)
     {
         _db = db;
         _fieldCodeOptionsService = fieldCodeOptionsService;
+        _identity = identity;
     }
 
-    public async Task<FieldCodeDto> SaveFieldCode(FieldCodeDto fieldCodeDto)
+    public async Task<FieldCodeDto> CreateFieldCode(FieldCodeDto fieldCodeDto)
     {
+        if (_identity.IsSupport == false)
+            throw new CustomException("Unauthorized Access.");
+
         if (fieldCodeDto.Id != 0)
         {
             return await UpdateFieldCode(fieldCodeDto);
         }
```

Comment:
```suggestion
```

We should not update in a create method

Reply:
I know we have a single button to save and update so should we change it up on the front-end as well.